src/pages/BooksPage.js

```diff
@@ -1,78 +1,92 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import PropTypes from 'prop-types';
 import {Link} from 'react-router-dom';
-
 import {connect} from 'react-redux';
-import {storeBooks} from '../redux/actions/storeBooks';
-
 import axios from 'axios';
 
+import {withDistance} from '../helpers/utils';
+import {storeBooks} from '../redux/actions/storeBooks';
+import {storeCoords} from '../redux/actions/storeCoords';
+import * as ROUTES from '../constants/routes';
+
 import Loader from '../components/Loader/Loader';
 import SearchResults from '../components/Books/SearchResults';
-import BooksFilters from '../components/Books/BooksFilters';
-
+import FilterGroup from '../components/Books/FilterGroup';
 import {index} from '../components/Algolia';
 
-import * as ROUTES from '../constants/routes';
-
 const _ = require('lodash/core');
 
 const BooksPage = props => {
-  const [borrowFilterOn, setBorrowFilterOn] = useState(false);
-  const [haveFilterOn, setHaveFilterOn] = useState(false);
+  const initialCoords = {lat: 0, lng: 0};
 
   const onFilter = filter => {
     index
       .search({
-        filters: `type:"${filter}"`
+        filters: filter
       })
       .then(res => {
-        props.storeBooks(res.hits);
+        props.storeBooks(withDistance(res.hits, props.coords));
       })
       .catch(err => {
         console.error(err);
       });
   };
 
-  const onBorrowFilter = e => {
-    !borrowFilterOn ? onFilter('to borrow') : getBooks();
-    setBorrowFilterOn(!borrowFilterOn);
-  };
-
-  const onHaveFilter = e => {
-    !haveFilterOn ? onFilter('to have') : getBooks();
-    setHaveFilterOn(!haveFilterOn);
-  };
-
   const getBooks = () => {
     axios({
       method: 'GET',
       url: `https://us-central1-bookio.cloudfunctions.net/getBooks`
     })
       .then(res => {
-        props.storeBooks(res.data);
+        props.storeBooks(withDistance(res.data, props.coords));
       })
       .catch(err => {
         console.error(err);
       });
   };
 
-  if (_.isEmpty(props.books) && props.searchBool === false) {
-    getBooks();
+  if (
+    _.isEmpty(props.books) &&
+    props.hasSearched === false &&
+    JSON.stringify(props.coords) !== JSON.stringify(initialCoords)
+  ) {
+    navigator.geolocation.getCurrentPosition(pos => {
+      props.storeCoords({lat: pos.coords.latitude, lng: pos.coords.longitude});
+      getBooks();
+    });
```

Comment:
why don't you get the location from the redux state?

Reply:
Done!